Polyfills/AbortController/Source/AbortController.cpp

```diff
@@ -26,14 +26,29 @@ namespace Babylon::Polyfills::Internal
 
     void AbortController::Abort(const Napi::CallbackInfo&)
     {
-        m_signal.Set("aborted", true);
-        m_signal.Get("onabort").As<Napi::Function>().Call({});
+        AbortSignal* sig = AbortSignal::Unwrap(m_signal.Value());
+
+        if (sig != nullptr)
+        {
+            sig->Abort();
+        }
+        else
+        {
+            throw std::runtime_error("AbortSignal should not be null");
+        }
     }
 
     AbortController::AbortController(const Napi::CallbackInfo& info)
         : Napi::ObjectWrap<AbortController>{info}
     {
         m_signal = Napi::Persistent(info.Env().Global().Get(AbortSignal::JS_ABORT_SIGNAL_CONSTRUCTOR_NAME).As<Napi::Function>().New({}));
+
+        const auto& sig = m_signal.Value();
+
+        if (sig.IsNull() || sig.IsUndefined())
+        {
+            throw std::runtime_error("AbortSignal should not be null/undefined");
+        }
```

Comment:
Why is this necessary?

Reply:
It probably isn't. It was brought up the other day asking if m_signal could be null/undefined but if anything in it's construction fails Napi would probably throw something first.

Reply:
I don't think it's possible to persist a null or undefined.